README.md

```diff
@@ -127,13 +127,73 @@ First-Party Sets is a proposal to standardize a mechanism that solves this issue
 
 # Applications
 
-In support of the various browser privacy models, first-party sets only control when embedded content that would otherwise be considered third-party can access its own state. Examples:
+Various applications of First-Party Sets have been suggested by different parties. The list below is an effort to compile suggested applications. 
```

Comment:
It might pay to talk about who gets what here.

What are the benefits that each constituency see as the result of doing FPS?  Users?  Sites?  Browsers?

Reply:
I disagree that the benefits to the users are indirect only. Each of these applications materially improves user's experience with the site. Specifically

1. Improved ease of use with applications like single sign on, unified consent management, persistent personalization etc.  
2. Improved security on sites with user uploaded content. 
3. Improved performance and reduced latency with caching etc. especially for users who have a slow connection to the internet.

Reply:
These are double-edged benefits.  


1. A site might be able to offer a smoother sign-on process, but this comes at a cost to users in a loss of control over how their activity on different sites is shared.  Similar things apply to consent (if I let one site access my camera, maybe I don't want other sites to gain the same privilege).  And personalization... the same argument is made in favour of tracking.
2. This is very much an indirect benefit.
3. I might concede this, but it's not that significant relative to the above.

Reply:
I can see how 3 can be detrimental to security by directly removing a security boundary, but I don't see how FPS can be beneficial to security. Where is that explained below? It's already the case that if you have user uploaded content you need to use a separate registrable domain to host that. I don't see how FPS changes anything there.

Reply:
I think the point here is that some user uploaded content is still access-restricted through cookies, which seems impossible to accomplish with full cross-site cookie blocking. The user content would have to be uploaded on the main origin to achieve effective access control. With the Same-Party cookie attribute proposal powered by FPS it would be possible to set a Same-Party cookie for this purpose (with attention to the security considerations outlined in #85).

I don't think this is an *indirect* benefit or use case of SameParty. This pattern is common with interactive coding environments, cloud storage providers and blogging/commenting platforms. Firefox has already [seen](https://bugzilla.mozilla.org/show_bug.cgi?id=1620530) [instances](https://bugzilla.mozilla.org/show_bug.cgi?id=1725216) [of](https://bugzilla.mozilla.org/show_bug.cgi?id=1664709) [breakage](https://bugzilla.mozilla.org/show_bug.cgi?id=1668408) caused by variations of this pattern (user content hosted on a separate authenticated site) and I'd expect to see many more. Some of these can be worked around with partitioning because they're limited to viewing content in an embedded fashion, but some other things are much harder to do.

Regarding the consent use case, I think Harneet was not referring to permissions and rather things like cookie consent management data which could be shared between sites of the same data controller, IIUC.

Reply:
Doesn't that presume a very specific outcome of the storage access discussion?  That is, if you block cookies in a cross-site context, this is one of the problems you create.  Solving it using FPS is - at least in my view - the wrong answer.  The right answer is to allow cross-site content access to cookies (storage might suffice, but you probably want to avoid extra round trips).

Reply:
I agree, the question is how do we relax cookies in this case.

None of the possible solutions is without drawbacks. Partitioning probably works for many cases but fails when moving out of an embedded context. Storage Access API mediates full 3PC access but might be too unwieldy and noisy for this, I suspect, rather popular use case. SameParty arguably allows for building/maintaining the best user experience while staying within the privacy boundaries guaranteed by FPS. However, these boundaries are based on ownership/data controllership (TBD) and not on whats-in-the-url-bar, which I think is a fair concern to consider given the historic security/privacy model of the web.

I'm going slightly off-topic here, so coming back to the original question I mainly wanted so show that solving the problem of separate (authenticated) user content sites is definitely serving a legitimate need of users (and developers).

Finally, your feedback makes me think that we should collectively do a better job of listing and evaluating the various known scenarios that will be broken by e.g. third-party cookie deprecation or navigational tracking protections, to have more common ground when discussing potential solutions.

Reply:
> given the historic security/privacy model of the web
Hm, I guess this is inaccurate in that the actual historic privacy model has been unregulated third-party access and the newly proposed ideal end state is "partitioned by whats-in-the-url-bar", which even where it's already implemented isn't fully achieved because of heuristic loop holes etc.